server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -468,12 +542,16 @@ public void sort() {
           index += batchHeaderLen + compressedSize;
           paddingBuf.clear();
           paddingBuf.limit(compressedSize);
-          readFully(originFileChannel, paddingBuf, originFile.getAbsolutePath());
+          if (isHdfs) {
+            readStreamFully(hdfsOriginInput, headerBuf, originFilePath);
```

Comment:
Write back to hdfs will not benefit from pagecache, maybe it's unnecessary to read fully here, we should do experiments later